storage/badger/monotonous_consumer_progress.go

```diff
@@ -0,0 +1,57 @@
+package badger
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/module/counters"
+	"github.com/onflow/flow-go/storage"
+)
+
+var _ storage.ConsumerProgress = (*MonotonousConsumerProgress)(nil)
+
+// MonotonousConsumerProgress represents the consumer progress with strict monotonous counter.
+type MonotonousConsumerProgress struct {
```

Comment:
rather than creating a monotonic `ConsumerProgress`, how about creating a `ConsumerProgress` backed monotonic counter?

The distinction is mainly in the API exposed. This API is fairly coupled with the jobqueue is doesn't make a lot of sense outside of that context. For instance, the consumer progress uses a get-then-initialize approach. To implement that here, you'll have to juggle setting the counter in multiple places and add extra error handling to ensure it set before use.

The new API could allow initializing/loading the initial value in the constructor instead of relying on the caller to do that and handling the error conditions associated with that not happening properly. It could also have more generic naming like `Store` and `Load`.

Reply:
[fbf6033](https://github.com/onflow/flow-go/pull/5752/commits/fbf60338a398d604b838e816bd650e7c4ffc7acf)